### PR TITLE
Added db file storage

### PR DESCRIPTION
**Pull request by Mixer57**

### CODE REVIEW

File: src/OnlineSales/DTOs/FileCreateDto.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// </copyright>
+
+using System.ComponentModel.DataAnnotations;
+
+namespace OnlineSales.DTOs;
+
+public class FileCreateDto
+{
+    [Required]
+    public IFormFile? File { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/DTOs/FileCreateDto.cs:12]: It must be possible to customise max possible files size similar to how it is done in MediaExtensionAttribute.

File: src/OnlineSales/Controllers/StorageController.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+                throw new InvalidModelStateException(ModelState);
+            }
+
+            using var fileStream = fileCreateDto.File.OpenReadStream();
+            var fileInBytes = new byte[incomingFileSize];
+            fileStream.Read(fileInBytes, 0, (int)fileCreateDto.File.Length);
+
+            var scopeAndFileExists = await pgDbContext!.StoredFiles!.Where(e => e.ScopeUid == fileCreateDto.ScopeUid && e.Name == incomingFileName!).ToArrayAsync();
+            if (!scopeAndFileExists.Any())
```
Comment (peterliapin) [inline src/OnlineSales/Controllers/StorageController.cs:57]: What happens if there is file with the same ScopeUid and Name in the database?

**Review by peterliapin — CHANGES_REQUESTED** [review]
Please review all the comments above and fix.

File: src/OnlineSales/Controllers/StorageController.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            {
+                StoredFile uploadedMedia = new()
```
[the remark this answers is not in the document]
Comment (Mixer57) [reply src/OnlineSales/Controllers/StorageController.cs:59]: var variable = new File() .... 
File variable = new() .. 
is the same

File: src/OnlineSales/Controllers/StorageController.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    {
+        private readonly PgDbContext pgDbContext;
+
+        public StorageController(PgDbContext pgDbContext)
+        {
+            this.pgDbContext = pgDbContext;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(typeof(MediaDetailsDto), StatusCodes.Status201Created)]
```
[the remark this answers is not in the document]
Comment (Mixer57) [reply src/OnlineSales/Controllers/StorageController.cs:30]: Fixed

File: src/OnlineSales/Controllers/StorageController.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mixer57) [reply src/OnlineSales/Controllers/StorageController.cs:59]: renamed

[the remark this answers is not in the document]
Comment (Mixer57) [reply src/OnlineSales/Controllers/StorageController.cs:59]: but rewrited

File: src/OnlineSales/DTOs/FileCreateDto.cs (hunk 1 above)
Comment (Mixer57) [reply src/OnlineSales/DTOs/FileCreateDto.cs:12]: FileExtensionAttribute added

File: src/OnlineSales/Controllers/StorageController.cs (hunk 2 above)
Comment (Mixer57) [reply src/OnlineSales/Controllers/StorageController.cs:57]: Fixed by adding check for exist record and update it when entity exists

**Pull request closed without merging**